mg/deps.go

```diff
@@ -95,18 +103,25 @@ func changeExit(old, new int) int {
 	return 1
 }
 
-func addDep(f interface{}) *onceFun {
-	var fn func() error
+func addDep(ctx context.Context, f interface{}) *onceFun {
+	var fn func(context.Context) error
 	switch f := f.(type) {
 	case func():
-		fn = func() error { f(); return nil }
+		fn = func(ctx context.Context) error { f(); return nil }
 	case func() error:
+		fn = func(ctx context.Context) error { return f() }
+	case func(context.Context):
+		fn = func(ctx context.Context) error { f(ctx); return nil }
+	case func(context.Context) error:
 		fn = f
+	default:
+		panic("attempted to add a dep that did not match required type")
```

Comment:
can you slap a 
`// should be impossible, since we already checked this`
above this?